src/main/resources/io/jenkins/plugins/designlibrary/Dialogs/index.jelly

```diff
@@ -0,0 +1,76 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:l="/lib/layout" xmlns:s="/lib/samples" xmlns:f="/lib/form">
+  <s:sample>
+    <p class="jdl-leading-paragraph">${%description}</p>
+    <template id="demo-template" data-title="${%Modal}">
+      <div>
+        <h2 class="jenkins-dialog__subtitle">${%Behaviour}</h2>
+        ${%Click on the "X" in the top right or outside to close me}.
+        <h2 class="jenkins-dialog__subtitle jenkins-!-margin-top-3">${%You can include icons}</h2>
+        <l:icon src="symbol-chatbox-outline plugin-ionicons-api" class="icon-sm"/> ${%Dialogs are cool}.
+      </div>
+    </template>
+    <div id="demo-form" class="jenkins-hidden" data-title="${%Order your ice cream}">
+      <f:form action="form" method="post">
+        <f:entry field="name" title="${%Name}">
+          <f:textbox/>
+        </f:entry>
+        <f:entry field="quantity" title="${%Quantity}">
+          <f:number min="1" max="9"/>
+        </f:entry>
+        <f:entry title="${%Choose flavor}">
+          <f:select field="flavor"/>
+        </f:entry>
+      </f:form>
+    </div>
+
+    <l:tabPane title="${%Alerts}">
+        <p class="jdl-paragraph">${%alerts}</p>
+        <s:code language="java" file="alert.js" code="showAlert()" executable="true"/>
+      </l:tabPane>
+      <l:tabPane title="${%Prompts}">
+        <p class="jdl-paragraph">${%prompts}</p>
+        <s:code language="java" file="prompt.js" code="showPrompt()" executable="true"/>
+      </l:tabPane>
+      <l:tabPane title="${%Confirmations}">
+        <p class="jdl-paragraph">${%confirmations}</p>
+        <s:code language="java" file="confirm.js" code="showConfirm()" executable="true"/>
+      </l:tabPane>
+      <l:tabPane title="${%Modals}">
+        <p class="jdl-paragraph">${%modals}</p>
+        <s:code language="java" file="modal.js" code="showModal()" executable="true"/>
+        <s:code language="xml" file="modal.jelly"/>
+      </l:tabPane>
+    <l:tabPane title="${%Forms}">
+      <p class="jdl-paragraph">${%forms}</p>
+      <br/><br/>
+      <f:toggleSwitch id="formsubmit" checkedTitle="${%Submit form}" checked="true"
+                      title="${%Handle form in javascript}."
+      />
+      <s:code language="java" file="form.js" code="showForm()" executable="true"/>
+      <s:code language="xml" file="form.jelly"/>
+    </l:tabPane>
+    <hr/>
+    <div>
+      <h2>Customizing the appearance and behaviour</h2>
+      All dialogs take a second optional parameter with options that allow to change certain aspects.
+      <ul>
+        <li><code>title</code>: Adds a title to the dialog</li>
+        <li><code>okText</code>: Adjust the text of the OK button, defaults to <code>OK</code> or <code>Yes</code></li>
```

Comment:
The default depends on the context. It is as of now `Yes` for confirm, `Submit` for forms and `OK` otherwise. Maybe that info should be moved to the corresponding dialog type page.